src/main/java/cn/nukkit/level/generator/object/tree/ObjectBirchTree.java

```diff
@@ -4,13 +4,15 @@
 import cn.nukkit.block.BlockWood;
 import cn.nukkit.level.ChunkManager;
 import cn.nukkit.math.NukkitRandom;
+import java.util.Random;
 
 /**
  * author: MagicDroidX
  * Nukkit Project
  */
 public class ObjectBirchTree extends ObjectTree {
-    protected int treeHeight = 7;
+    Random random = new Random();
+    protected int treeHeight = random.nextInt(8-6+1)+8;
```

Comment:
@TheDiamondYT1 This branch is tested and work perfectly. All is from singleplayer *values* and works VERY VERY fine.